lib/timsort_lib.c

```diff
@@ -152,10 +138,15 @@ size_t min(size_t a, size_t b)
  */
 void timSort_classic(int arr[], size_t length)
 {
+	if (length <= 1)
```

Comment:
Isn't this just `if (length == 0)` in disguise?

Reply:
No. There are two types of lists that are always sorted: empty lists ( length == 0 ) and singleton lists ( length == 1) hence the ` length <= 1` predicate.

Reply:
Ah, I see.